swarm/storage/mru/resource.go

```diff
@@ -362,57 +326,49 @@ func (self *Handler) GetVersion(nameHash string) (uint32, error) {
 }
 
 // \TODO should be hashsize * branches from the chosen chunker, implement with FileStore
-func (self *Handler) chunkSize() int64 {
+func (h *Handler) chunkSize() int64 {
 	return chunkSize
 }
 
-// Creates a new root entry for a mutable resource identified by `name` with the specified `frequency`.
-//
-// The signature data should match the hash of the idna-converted name by the validator's namehash function, NOT the raw name bytes.
-//
+// New creates a new metadata chunk for a Mutable Resource identified by `name` with the specified `frequency`.
+// It uses the public key from the Signer registered with the Handler object
 // The start block of the resource update will be the actual current block height of the connected network.
-func (self *Handler) New(ctx context.Context, name string, frequency uint64) (storage.Address, *resource, error) {
+func (h *Handler) New(ctx context.Context, name string, frequency uint64) (storage.Address, *resource, error) {
+	return h.NewWithPublicKey(ctx, name, frequency, h.signer.PublicKey())
+}
+
+// NewWithPublicKey performs the same action as New, but enables a custom public key to be used for the Mutable Resource
+func (h *Handler) NewWithPublicKey(ctx context.Context, name string, frequency uint64, publicKey *ecdsa.PublicKey) (storage.Address, *resource, error) {
 
 	// frequency 0 is invalid
 	if frequency == 0 {
-		return nil, nil, NewError(ErrInvalidValue, "Frequency cannot be 0")
+		return nil, nil, NewError(ErrInvalidValue, "frequency cannot be 0")
```

Comment:
just a thought: maybe 0 should not be invalid, just indicate that only versions are used?

Reply:
Might be a good idea. But let's save that one for later.